src/main/java/network/brightspots/rcv/GuiTabulateController.java

```diff
@@ -0,0 +1,381 @@
+        ? guiConfigController.saveFile(tempSaveButton, true)
+        : savedConfigFilePath;
+
+    ContestConfig config = ContestConfig.loadContestConfig(path);
+    configOutputPath = config.getOutputDirectory();
+    if (!configOutputPath.endsWith("/")) {
+      configOutputPath += "/";
+    }
+
+    return path;
+  }
+
+  private boolean isConfigSavedOrTempFileReady() {
+    return savedConfigFilePath != null || useTemporaryConfigBeforeTabulation;
+  }
+
+  private void setTabulationButtonStatus() {
+    if (isConfigSavedOrTempFileReady()) {
+      // Don't override the progress text unless we're past the Save stage
+      updateGuiWithNameEnteredStatus();
+    }
+
+    if (isConfigSavedOrTempFileReady() && !userNameField.getText().isEmpty()) {
+      enableButtonsUpTo(loadCvrButton);
+    } else {
+      enableButtonsUpTo(null);
+    }
+  }
+
+  private void updateGuiNotifyConfigSaved() {
+    filepath.setStyle(filledFieldStyle);
+    tempSaveButton.setStyle(filledFieldStyle);
+    saveButton.setStyle(filledFieldStyle);
+    tempSaveButton.setDisable(true);
+    saveButton.setDisable(true);
+    updateProgressText();
+  }
+
+  private void updateGuiWithNameEnteredStatus() {
+    if (userNameField.getText().isEmpty()) {
+      userNameField.setStyle(unfilledFieldStyle);
+    } else {
+      userNameField.setStyle(filledFieldStyle);
+    }
+  }
+
+  private void initializeSaveButtonStatuses() {
+    filepath.setText(guiConfigController.getSelectedFilePath());
+    filepath.positionCaret(filepath.getLength() - 1); // scroll to end
+    filepath.setStyle(unfilledFieldStyle);
+    saveButton.setStyle(unfilledFieldStyle);
+    tempSaveButton.setStyle(unfilledFieldStyle);
+
+    GuiConfigController.ConfigComparisonResult result = guiConfigController.compareConfigs();
+    switch (result) {
+      case DIFFERENT:
+        tempSaveButton.setVisible(false);
+        break;
+      case DIFFERENT_BUT_VERSION_IS_TEST:
+        tempSaveButton.setVisible(true);
+        break;
+      case SAME:
+        saveButton.setText("Saved!");
+        savedConfigFilePath = guiConfigController.getSelectedFilePath();
+        tempSaveButton.setVisible(false);
+        updateGuiNotifyConfigSaved();
+        break;
+      default:
+        throw new IllegalStateException("Unexpected value: " + result);
+    }
+  }
+
+  private void updateProgressText() {
+    if (!isConfigSavedOrTempFileReady()) {
+      progressText.setText("Save the config file to continue.");
+    } else if (userNameField.getText().isEmpty()) {
+      progressText.setText("Please enter your name to continue.");
+    } else {
+      progressText.setText("");
+    }
+  }
+
+  private void openOutputDirectoryInFileExplorer() {
+    String[] cmd = new String[]{"open", "-R", configOutputPath};
+
+    try {
+      Runtime.getRuntime().exec(cmd, null);
+    } catch (IOException e) {
+      Logger.warning("Failed to open file explorer: " + e.getMessage());
```

Comment:
Perhaps -- I'll spin up my Windows machine and see if I can reproduce. It should happen on develop too, since there are no logging changes in this PR

Reply:
It's totally fine to spin this off into a separate issue, linking to this comment thread. CC: @yezr

Reply:
created https://github.com/BrightSpots/rcv/issues/840 as a P2. P2 since I'm assuming we won't see this issue anymore since you fixed the underlying problem? But do agree that looking into whatever reason could cause the log to hang should be investigated and squashed once and for all.

Reply:
Thanks for filing this. I responded in the bug, as the second part still stands, but agree it's probably not super high priority.